hoot-core/src/main/cpp/hoot/core/algorithms/WayJoiner.cpp

```diff
@@ -0,0 +1,337 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018, 2019 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "WayJoiner.h"
+
+//  Hoot
+#include <hoot/core/criterion/AreaCriterion.h>
+#include <hoot/core/criterion/OneWayCriterion.h>
+#include <hoot/core/elements/NodeToWayMap.h>
+#include <hoot/core/index/OsmMapIndex.h>
+#include <hoot/core/io/OsmMapWriterFactory.h>
+#include <hoot/core/ops/RecursiveElementRemover.h>
+#include <hoot/core/ops/ReplaceElementOp.h>
+#include <hoot/core/schema/TagMergerFactory.h>
+
+#include <unordered_set>
+#include <vector>
+
+using namespace std;
+
+namespace hoot
+{
+
+WayJoiner::WayJoiner()
+    : _leavePid(false),
+      _numJoined(0)
+{
+}
+
+void WayJoiner::join(const OsmMapPtr& map)
```

Comment:
All of the functions now in `WayJoiner` come directly from `WayJoiner1` but are inherited and that is why `WayJoiner1` is mostly empty.